lib/SIL/SILLocation.cpp

```diff
@@ -20,8 +20,14 @@
 
 using namespace swift;
 
+// 64-bit is 24 bytes, 32-bit is 20 bytes.
+#ifdef __LP64__
```

Comment:
I'll go ahead and make that change for the ifdef tomorrow (Monday). I think the only way you can deflate this type any more is to remove stuff out of the union so it can shrink. There's no padding on 32-bit x86 at least, and ARM shouldn't be any different.

I think the only concern with the macro is that at least as of a few days ago, it wasn't used anywhere in the swift project. So I guess it's more of a question of: is this where swift wants to go all-up as their macro for checking for 64-bit in the future? I personally think being able to find the various places 64-bit is explicitly checked for is valuable, so I'm a little hesitant to fracture that. 

But I like the general idea.

Reply:
I looked around and LP64 isn't really being used either. I think we just haven't had anyone really put effort into building a 32-bit version of the compiler.

(In the runtime we use `__POINTER_WIDTH__ == 64`, but that's because the runtime is built with the same Clang that's embedded in the Swift compiler.)

Reply:
I agree it isn't used in many places, but 8 is still not zero. I agree with the suggestion, I just want to make sure this is also a statement of intent for the future too, is all.

----

What got me looking at this stuff in the first place is that there are 3.1.1 binaries for the Raspberry Pi floating around that work quite well and are stable. But with folks moving to support only 4.x, it's making it hard to stick with 3.1.1. 

4.1 has even been mostly working so long as you didn't use Foundation for anything, because of a nasty bug lurking there (we've got the community binaries unblocked, but still need to re-fix the master branch itself).

I'm mostly trying to get things into a state where the community isn't facing the daunting prospect of dealing with regressions every major release, which lags adoption of new versions by half a year or more. Mostly because I think Swift is working really well on these single-board computers as an alternative to Node or Python. It just really needs some stability, and this is a step towards that.

Reply:
Again, you have to distinguish the uses in the runtime and tests (target conditions) from the uses in the compiler (host conditions). The only one that's in the compiler is the one that's `defined(__BIG_ENDIAN__) && defined(__LP64__)`, and I'm not even sure that condition is correct anyway.

Reply:
That's fair, although I'd imagine you'd probably not want too much LP64 leakage there either?

That big endian condition is "correct" in the sense that you have to get pretty niche (more so than the single board computers) to make it fail.

Reply:
I think my concern is that that should have been a test of the target platform as well, rather than the host. If anyone is cross-compiling from a big-endian 64-bit compiler to a 32-bit anything, something funny's going to happen here.

Reply:
I've redone the assert, this time just being more explicit about the underlying types. The #ifdef also works, but this has the same effect without an ifdef. It's also a bit easier to understand how the byte size was reached and correlate it with the SILLocation class' structure.

Reply:
All right, that works. Thanks!